tests/test_enterprise/api/test_views.py

```diff
@@ -5503,3 +5503,60 @@ def test_same_enable_universal_link(self):
         self.assertEqual(response.status_code, 200)
         response = response.json()
         self.assertEqual(response['detail'], 'No changes')
+
+
+@mark.django_db
+class TestPlotlyAuthView(APITest):
+    """
+    Test PlotlyAuthView
+    """
+
+    PLOTLY_TOKEN_ENDPOINT = 'plotly-token'
+
+    def setUp(self):
+        """
+        Common setup for all tests.
+        """
+        super().setUp()
+        self.client.login(username=self.user.username, password=TEST_PASSWORD)
+        self.enterprise_uuid = fake.uuid4()
+        self.enterprise_uuid2 = fake.uuid4()
+        self.url = settings.TEST_SERVER + reverse(
+            self.PLOTLY_TOKEN_ENDPOINT, kwargs={'enterprise_uuid': self.enterprise_uuid}
+        )
+
+    def test_view_with_normal_user(self):
+        """
+        Verify that a user without having `enterprise.can_access_admin_dashboard` role can't access the view.
+        """
+        response = self.client.get(self.url)
+        assert response.status_code == status.HTTP_403_FORBIDDEN
+        assert response.json() == {'detail': 'Missing: enterprise.can_access_admin_dashboard'}
+
+    def test_view_with_admin_user(self):
+        """
+        Verify that an enterprise admin user having `enterprise.can_access_admin_dashboard` role can access the view.
+        """
+        self.set_jwt_cookie(ENTERPRISE_ADMIN_ROLE, self.enterprise_uuid)
```

Comment:
nit: we can move `set_jwt_cookie` to `setUp` and rename `self.enterprise_uuid` as `admin_enterprise_uuid`
`self.enterprise_uuid2` to `self.non_admin_enterprise_uuid`

Reply:
but we don't want to set the jwt cookie for the normal user in `test_view_with_normal_user`